src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/processor/CatalogItemProcessor.java

```diff
@@ -198,7 +198,9 @@ public T process(T resource) throws Exception {
                                     .getResourceByName(((ResourceInfo) resource).getName(), clz);
                     if (source != null && source.getStore() != null) {
                         StoreInfo ds =
-                                getCatalog().getStoreByName(source.getStore().getName(), clz);
+                                getCatalog()
```

Comment:
This one gave me pause, what's happening here? The old code did not make much sense TBH, but I would have expected to see CoveageStoreInfo for coverages and DataStoreInfo for rasters. Then again, I believe one cannot give the same name to two stores, so it's probably fine.

Reply:
Indeed that looks like a bug, although for the sake of this PR it'd be out of scope. I did fix it in a separate PR though, see #4893. A `ResourceInfo` subclass was being used to ask for a `StoreInfo`. That method suffered a lot more in terms of readability, so I tried to make it better in that pr.